src/libs/ReportUtils.ts

```diff
@@ -5372,12 +5372,14 @@ function canUserPerformWriteAction(report: OnyxEntry<Report>) {
 /**
  * Returns ID of the original report from which the given reportAction is first created.
  */
-function getOriginalReportID(reportID: string, reportAction: OnyxEntry<ReportAction>): string | undefined {
+function getOriginalReportID(reportID: string, reportAction: OnyxEntry<ReportAction>, shouldReturnTransactionThread = false): string | undefined {
     const reportActions = allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`];
     const currentReportAction = reportActions?.[reportAction?.reportActionID ?? ''] ?? null;
-    const transactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(reportID, reportActions ?? ([] as ReportAction[]));
-    if (transactionThreadReportID !== null) {
-        return Object.keys(currentReportAction ?? {}).length === 0 ? transactionThreadReportID : reportID;
+    if (shouldReturnTransactionThread) {
```

Comment:
why is `transactionThreadReportID` computation blocked under this flag? Shouldn't it break for calls of  `getOriginalReportID` since only one place we are sending `true` and others will be false by default?

And the change here is different than your original proposal https://github.com/Expensify/App/issues/36057#issuecomment-1932681036, could you provide information on how this change is expected to fix the issue?

@FitseTLT

Reply:
@abdulrahuman5196 I have already commented an explanation for both in Details section on the PR description.

Reply:
@FitseTLT 
This is the from the details section

>The expected behaviour has changed from OP of the issue so now we navigate to the parent report and highlight on the parent message according to this https://github.com/Expensify/App/issues/36057#issuecomment-2059895416

This is the expected behaviour I get from the issue, but not sure on how it is being fixed with this PR. Because the code change in the proposal is different from the change in this PR. https://github.com/Expensify/App/issues/36057#issuecomment-1932681036. Do let me know if there was a different discussion.

>I also fixed a regression from
https://github.com/Expensify/App/pull/39474 which caused reportAction used in context menu to be null as it passes the wrong originalReportID when IOU type report action is displayed as the parent message in a thread.

What is the UX regression on this issue?, to check if its fixed? I don't see any regression statement in that PR as well its relevant GH. Do add those in test steps if its a specific case.

Reply:
>This is the expected behaviour I get from the issue, but not sure on how it is being fixed with this PR. Because the code change in the proposal is different from the change in this PR. https://github.com/Expensify/App/issues/36057#issuecomment-1932681036. Do let me know if there was a different discussion.

https://github.com/Expensify/App/issues/36057#issuecomment-2059895416 is where the expectation changed that's why the pr is different from the proposal.

>What is the UX regression on this issue?, to check if its fixed? I don't see any regression statement in that PR as well its relevant GH. Do add those in test steps if its a specific case.

1. On a collect workspace: Create a money request and ensure it is the only transaction in the expense report
2. On the transaction thread send a message
3. "Reply in thread" with it
4. Now on the new thread right click on the request preview parent message and copy link
5. Paste the link 

Because of the regression caused from https://github.com/Expensify/App/pull/39474 the `reportActionID` part of the link will be `undefined`. But the problem it caused is wider as it makes the all context menu operations not to work (like copy to clipboard )